File: services/device/examples/persistent_message_queue.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
PersistentMessageQueue - 本地持久化消息队列（Python版本）

核心功能：
  ✅ 本地JSON文件持久化（断电不丢、重启不丢）
  ✅ 原子写入操作（防止数据损坏）
  ✅ 消息状态管理（PENDING → SENT → ACKED）
  ✅ 自动清理已确认消息
  ✅ 按时间顺序保证（FIFO）
  ✅ 最大容量限制（防止磁盘占满）

使用示例：
    from persistent_message_queue import PersistentMessageQueue

    queue = PersistentMessageQueue(storage_path='./data/queue.json')

    # 入队消息
    result = queue.enqueue({'type': 'status', 'data': {'battery': 85}})

    # 获取待发消息
    pending = queue.get_pending_messages()

    # 标记为已发送
    queue.mark_as_sent(message_id)

    # 收到ACK确认
    queue.mark_as_acked(message_id)
"""

import json
import os
import uuid
import time
import threading
from typing import Optional, Dict, Any, List
from dataclasses import dataclass, asdict, field
from enum import Enum, auto


class MessageStatus(Enum):
    """消息状态枚举"""
    PENDING = "PENDING"      # 待发送
    SENT = "SENT"            # 已发送，等待ACK
    ACKED = "ACKED"          # 已确认（可删除）


@dataclass
class QueueMessage:
    """队列中的消息对象"""
    id: str
    type: str
    payload: Dict[str, Any]
    status: str
    created_at: float
    sent_at: Optional[float] = None
    acked_at: Optional[float] = None
    retry_count: int = 0
    last_error: Optional[str] = None
    metadata: Dict[str, Any] = field(default_factory=dict)


class PersistentMessageQueue:
    """
    本地持久化消息队列

    特性：
      - 基于JSON文件的持久化存储
      - 线程安全（支持多线程访问）
      - 原子写入（先写临时文件再重命名）
      - 自动保存机制
    """

    def __init__(self, options: Optional[Dict[str, Any]] = None):
        options = options or {}

        # 存储配置
        self.storage_path = options.get('storage_path', './data/device_message_queue.json')
        self.max_queue_size = options.get('max_queue_size', 10000)  # 最大10000条
        self.auto_save_interval = options.get('auto_save_interval', 5.0)  # 每5秒自动保存

        # 内部状态
        self.messages: List[QueueMessage] = []
        self.version = '1.0'
        self.created_at = ''
        self._dirty = False
        self._lock = threading.Lock()
        self._save_timer: Optional[threading.Timer] = None

        # 初始化
        self._ensure_storage_directory()
        self._load_from_file()

        # 启动自动保存定时器
        if self.auto_save_interval > 0:
            self._start_auto_save()

        print(f"[MessageQueue] 📦 持久化消息队列初始化完成")
        print(f"[MessageQueue] 📍 存储路径: {self.storage_path}")
        print(f"[MessageQueue] 📊 当前队列长度: {len(self.messages)}")

    def enqueue(self, message: Dict[str, Any]) -> Dict[str, Any]:
        """
        入队消息（写入本地持久化存储）

        参数：
            message: 消息数据字典

        返回：
            {
                'success': bool,
                'message_id': str,
                'queue_size': int,
                'error': str (可选)
            }
        """
        try:
            with self._lock:
                # 检查队列是否已满
                if len(self.messages) >= self.max_queue_size:
                    removed_id = self._remove_oldest_pending()
                    print(f"[MessageQueue] ⚠️  队列已满，移除最旧消息: {removed_id}")

                # 生成唯一消息ID
                message_id = str(uuid.uuid4())

                # 构造消息对象
                queue_msg = QueueMessage(
                    id=message_id,
                    type=message.get('type', 'unknown'),
                    payload=message.get('payload', message),
                    status=MessageStatus.PENDING.value,
                    created_at=time.time(),
                    retry_count=0,
                    metadata=message.get('metadata', {})
                )

                # 加入队列并排序
                self.messages.append(queue_msg)
                self._sort_messages()

                # 标记脏数据
                self._dirty = True

                result = {
                    'success': True,
                    'message_id': message_id,
                    'queue_size': len(self.messages)
                }

            print(f"[MessageQueue] ✅ 消息入队成功:")
            print(f"   ID: {message_id}")
            print(f"   类型: {queue_msg.type}")
            print(f"   队列长度: {len(self.messages)}")

            return result

        except Exception as error:
            print(f"[MessageQueue] ❌ 入队失败: {error}")
            return {
                'success': False,
                'error': str(error)
            }

    def get_pending_messages(self) -> List[Dict[str, Any]]:
        """获取所有待发送的消息（按时间顺序）"""
        with self._lock:
            pending = [m for m in self.messages if m.status == MessageStatus.PENDING.value]
            pending.sort(key=lambda x: x.created_at)
            return [asdict(m) for m in pending]

    def get_sent_messages(self) -> List[Dict[str, Any]]:
        """获取所有已发送但未确认的消息"""
        with self._lock:
            sent = [m for m in self.messages if m.status == MessageStatus.SENT.value]
            sent.sort(key=lambda x: x.sent_at or 0)
            return [asdict(m) for m in sent]

    def mark_as_sent(self, message_id: str) -> bool:
        """
        标记消息为已发送（等待ACK）

        参数：
            message_id: 消息ID

        返回：
            bool: 是否成功标记
        """
        with self._lock:
            for msg in self.messages:
                if msg.id == message_id:
                    msg.status = MessageStatus.SENT.value
                    msg.sent_at = time.time()
                    msg.retry_count += 1
                    self._dirty = True

                    print(f"[MessageQueue] 📤 消息已标记为SENT: {message_id} "
                          f"(重试次数: {msg.retry_count})")
                    return True

            return False

    def mark_as_acked(self, message_id: str) -> bool:
        """
        收到ACK确认，从队列中移除消息

        参数：
            message_id: 消息ID

        返回：
            bool: 是否成功处理
        """
        with self._lock:
            for i, msg in enumerate(self.messages):
                if msg.id == message_id:
                    msg.status = MessageStatus.ACKED.value
                    msg.acked_at = time.time()

                    print(f"[MessageQueue] ✅ 消息已确认(ACK): {message_id}")

                    # 从队列中移除
                    self.messages.pop(i)
                    self._dirty = True
                    return True

            print(f"[MessageQueue] ⚠️  未找到消息ID: {message_id}")
            return False

    def mark_as_failed(self, message_id: str, error: str) -> bool:
        """
        标记消息发送失败（保持PENDING状态以便重试）

        参数：
            message_id: 消息ID
            error: 错误信息

        返回：
            bool: 是否成功标记
        """
        with self._lock:
            for msg in self.messages:
                if msg.id == message_id:
                    msg.status = MessageStatus.PENDING.value
                    msg.last_error = error
                    msg.sent_at = None  # 清除发送时间
                    self._dirty = True

                    print(f"[MessageQueue] ❌ 消息发送失败: {message_id}, 错误: {error}")
                    return True

            return False

    def get_all_messages(self) -> List[Dict[str, Any]]:
        """获取所有消息（用于调试）"""
        with self._lock:
            return [asdict(m) for m in self.messages]

    def get_stats(self) -> Dict[str, Any]:
        """获取队列统计信息"""
        with self._lock:
            pending = sum(1 for m in self.messages if m.status == MessageStatus.PENDING.value)
            sent = sum(1 for m in self.messages if m.status == MessageStatus.SENT.value)
            acked = sum(1 for m in self.messages if m.status == MessageStatus.ACKED.value)

            oldest_pending_time = self._get_oldest_pending_time()

            return {
                'total': len(self.messages),
                'pending': pending,
                'sent': sent,
                'acked': acked,
                'max_capacity': self.max_queue_size,
                'utilization': f"{(len(self.messages) / self.max_queue_size * 100):.1f}%",
                'oldest_pending': oldest_pending_time,
                'storage_path': self.storage_path
            }

    async def clear_all(self):
        """清空所有消息（慎用！）"""
        with self._lock:
            self.messages = []
            await self._save_to_file()
            print('[MessageQueue] 🗑️  队列已清空')

    async def force_save(self):
        """强制立即保存到文件"""
        await self._save_to_file()

    def destroy(self):
        """销毁实例（停止自动保存）"""
        if self._save_timer:
            self._save_timer.cancel()
            self._save_timer = None

        # 最后一次保存
        self.force_sync()

        print('[MessageQueue] 💀 实例已销毁')

    force_sync = lambda self: self._save_to_file_sync()

    # ══════════════════════════════════════════════
    # 私有方法
    # ══════════════════════════════════════════════

    def _sort_messages(self):
        """按时间排序消息"""
        self.messages.sort(key=lambda x: x.created_at)

    def _get_oldest_pending_time(self) -> str:
        """获取最旧待发消息的年龄"""
        pending = [m for m in self.messages if m.status == MessageStatus.PENDING.value]
        if pending:
            age_seconds = time.time() - min(pending, key=lambda x: x.created_at).created_at
            return f"{int(age_seconds)}s"
        return 'N/A'

    def _remove_oldest_pending(self) -> Optional[str]:
        """移除最旧的待发消息"""
        for i, msg in enumerate(self.messages):
            if msg.status == MessageStatus.PENDING.value:
                removed = self.messages.pop(i)
                self._dirty = True
                return removed.id
        return None

    def _ensure_storage_directory(self):
        """确保存储目录存在"""
        directory = os.path.dirname(self.storage_path)
        if directory and not os.path.exists(directory):
            os.makedirs(directory, exist_ok=True)
            print(f"[MessageQueue] 📁 创建存储目录: {directory}")

    def _load_from_file(self):
        """从文件加载队列"""
        try:
            if os.path.exists(self.storage_path):
                with open(self.storage_path, 'r', encoding='utf-8') as f:
                    data = json.load(f)

                self.version = data.get('version', '1.0')
                self.created_at = data.get('created_at', '')
                
                messages_data = data.get('messages', [])
                self.messages = [
                    QueueMessage(**msg_data) 
                    for msg_data in messages_data
                ]

                print(f"[MessageQueue] 📂 从文件加载队列: {len(self.messages)}条消息")
            else:
                print("[MessageQueue] 📝 存储文件不存在，创建新队列")
                self._save_to_file_sync()

        except Exception as error:
            print(f"[MessageQueue] ❌ 加载文件失败: {error}")
            print("[MessageQueue] ⚠️  使用空队列启动")

            # 备份损坏的文件
            if os.path.exists(self.storage_path):
                backup_path = f"{self.storage_path}.corrupted.{int(time.time())}"
                os.rename(self.storage_path, backup_path)
                print(f"[MessageQueue] 💾 损坏文件已备份到: {backup_path}")

    async def _save_to_file(self):
        """异步保存到文件"""
        if not self._dirty:
            return

        await self._do_save()

    def _save_to_file_sync(self):
        """同步保存到文件"""
        if not self._dirty:
            return

        # 在新线程中执行异步保存
        import asyncio
        try:
            loop = asyncio.get_event_loop()
            if loop.is_running():
                # 如果事件循环正在运行，使用线程池
                from concurrent.futures import ThreadPoolExecutor
                with ThreadPoolExecutor() as executor:
                    executor.submit(self._do_save_sync).result()
            else:
                loop.run_until_complete(self._do_save())
        except RuntimeError:
            # 无事件循环可用，直接同步保存
            self._do_save_sync()

    def _do_save(self):
        """实际执行保存操作"""
        if self._lock.locked():
            print("[MessageQueue] ⚠️  文件锁被占用，跳过保存")
            return

        with self._lock:
            try:
                data = {
                    'version': self.version,
                    'created_at': self.created_at,
                    'last_saved': time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime()),
                    'messages': [asdict(m) for m in self.messages],
                    'stats': self.get_stats()
                }

                json_str = json.dumps(data, indent=2, ensure_ascii=False)

                # 原子写入：先写临时文件再重命名
                temp_path = f"{self.storage_path}.tmp"
                with open(temp_path, 'w', encoding='utf-8') as f:
                    f.write(json_str)
                os.replace(temp_path, self.storage_path)

                self._dirty = False
                print(f"[MessageQueue] 💾 队列已保存: {len(self.messages)}条消息")

            except Exception as error:
                print(f"[MessageQueue] ❌ 保存失败: {error}")

                # 清理临时文件
                temp_path = f"{this.storage_path}.tmp"
                if os.path.exists(temp_path):
                    try:
                        os.remove(temp_path)
                        print("[MessageQueue] 🧹 已清理临时文件")
                    except Exception as e:
                        print(f"[MessageQueue] ❌ 清理失败: {e}")

    def _do_save_sync(self):
        """同步版本的实际保存操作"""
        self._do_save()

    def _start_auto_save(self):
        """启动自动保存定时器"""
        def save_loop():
            while True:
                time.sleep(self.auto_save_interval)
                if self._dirty:
                    self._save_to_file_sync()

        save_thread = threading.Thread(target=save_loop, daemon=True)
        save_thread.start()


# 使用示例
if __name__ == '__main__':
    print("\n" + "=" * 60)
    print("🧪 PersistentMessageQueue 测试演示")
    print("=" * 60 + "\n")

    # 创建队列实例
    queue = PersistentMessageQueue({
        'storage_path': './data/test_queue.json',
        'max_queue_size': 100
    })

    # 测试入队
    print("📍 测试1: 入队消息\n")
    for i in range(5):
        result = queue.enqueue({
            'type': 'status_report',
            'payload': {
                'battery_level': 80 + i,
                'timestamp': time.time(),
                'test_number': i + 1
            }
        })
        print(f"   结果: {result}\n")

    # 获取统计信息
    stats = queue.get_stats()
    print("\n📊 队列统计:")
    for key, value in stats.items():
        print(f"   {key}: {value}")

    # 获取待发消息
    print("\n📤 待发消息:")
    pending = queue.get_pending_messages()
    for msg in pending[:3]:  # 只显示前3条
        print(f"   ID: {msg['id'][:8]}... 类型: {msg['type']} 状态: {msg['status']}")

    # 标记为已发送
    print("\n✅ 标记为已发送:")
    if pending:
        first_msg_id = pending[0]['id']
        queue.mark_as_sent(first_msg_id)
        print(f"   消息 {first_msg_id[:8]}... → SENT")

    # 模拟收到ACK
    print("\n🎉 收到ACK确认:")
    acked = queue.mark_as_acked(first_msg_id)
    print(f"   ACK处理结果: {acked}")

    # 最终统计
    final_stats = queue.get_stats()
    print(f"\n📈 最终统计: 待发={final_stats['pending']}, 总计={final_stats['total']}")

    # 清理
    print("\n👋 清理测试数据...")
    queue.clear_all()
    queue.destroy()
    print("✅ 测试完成!\n")

File: services/device/examples/persistent-message-queue.js
/**
 * PersistentMessageQueue - 本地持久化消息队列
 *
 * 核心功能：
 *   ✅ 本地JSON文件持久化（断电不丢、重启不丢）
 *   ✅ 原子写入操作（防止数据损坏）
 *   ✅ 消息状态管理（PENDING → SENT → ACKED）
 *   ✅ 自动清理已确认消息
 *   ✅ 按时间顺序保证（FIFO）
 *   ✅ 最大容量限制（防止磁盘占满）
 *
 * 存储格式：
 *   {
 *     "version": "1.0",
 *     "created_at": "2026-05-20T22:00:00Z",
 *     "messages": [
 *       {
 *         "id": "uuid-xxx",
 *         "type": "status_report",
 *         "payload": {...},
 *         "status": "PENDING",        // PENDING | SENT | ACKED
 *         "created_at": 1700000000000,
 *         "sent_at": null,
 *         "acked_at": null,
 *         "retry_count": 0,
 *         "last_error": null
 *       }
 *     ]
 *   }
 */

const fs = require('fs');
const path = require('path');
const crypto = require('crypto');
const { v4: uuidv4 } = require('uuid');

class PersistentMessageQueue {
  constructor(options = {}) {
    // 存储配置
    this.storagePath = options.storagePath || './data/device_message_queue.json';
    this.maxQueueSize = options.maxQueueSize || 10000; // 最大10000条
    this.autoSaveInterval = options.autoSaveInterval || 5000; // 每5秒自动保存

    // 内部状态
    this.messages = [];
    this.version = '1.0';
    this.createdAt = new Date().toISOString();
    this._dirty = false; // 标记是否有未保存的更改
    this._saveTimer = null;
    this._lock = false; // 文件锁，防止并发写入

    // 初始化
    this._ensureStorageDirectory();
    this._loadFromFile();

    // 启动自动保存定时器
    if (this.autoSaveInterval > 0) {
      this._startAutoSave();
    }

    console.log(`[MessageQueue] 📦 持久化消息队列初始化完成`);
    console.log(`[MessageQueue] 📍 存储路径: ${this.storagePath}`);
    console.log(`[MessageQueue] 📊 当前队列长度: ${this.messages.length}`);
  }

  /**
   * enqueue - 入队消息（写入本地持久化存储）
   */
  async enqueue(message) {
    try {
      // 检查队列是否已满
      if (this.messages.length >= this.maxQueueSize) {
        const removed = this._removeOldestPending();
        console.warn(`[MessageQueue] ⚠️  队列已满，移除最旧的待发送消息: ${removed}`);
      }

      // 生成唯一消息ID
      const messageId = uuidv4();

      // 构造消息对象
      const queueMessage = {
        id: messageId,
        type: message.type || 'unknown',
        payload: message.payload || message,
        status: 'PENDING', // PENDING: 待发送, SENT: 已发送等待ACK, ACKED: 已确认
        created_at: Date.now(),
        sent_at: null,
        acked_at: null,
        retry_count: 0,
        last_error: null
      };

      // 加入队列（按时间排序）
      this.messages.push(queueMessage);
      this._sortMessages();

      // 标记脏数据，触发保存
      this._markDirty();

      console.log(`[MessageQueue] ✅ 消息入队成功:`);
      console.log(`   ID: ${messageId}`);
      console.log(`   类型: ${queueMessage.type}`);
      console.log(`   队列长度: ${this.messages.length}`);

      return {
        success: true,
        message_id: messageId,
        queue_size: this.messages.length
      };

    } catch (error) {
      console.error(`[MessageQueue] ❌ 入队失败:`, error);
      return {
        success: false,
        error: error.message
      };
    }
  }

  /**
   * getPendingMessages - 获取所有待发送的消息（按时间顺序）
   */
  getPendingMessages() {
    return this.messages
      .filter(msg => msg.status === 'PENDING')
      .sort((a, b) => a.created_at - b.created_at);
  }

  /**
   * getSentMessages - 获取所有已发送但未确认的消息
   */
  getSentMessages() {
    return this.messages
      .filter(msg => msg.status === 'SENT')
      .sort((a, b) => (a.sent_at || 0) - (b.sent_at || 0));
  }

  /**
   * markAsSent - 标记消息为已发送（等待ACK）
   */
  markAsSent(messageId) {
    const message = this.messages.find(m => m.id === messageId);
    if (message) {
      message.status = 'SENT';
      message.sent_at = Date.now();
      message.retry_count += 1;
      this._markDirty();

      console.log(`[MessageQueue] 📤 消息已标记为SENT: ${messageId} (重试次数: ${message.retry_count})`);
      return true;
    }
    return false;
  }

  /**
   * markAsAcked - 收到ACK确认，删除消息（或标记为ACKED）
   */
  markAsAcked(messageId) {
    const index = this.messages.findIndex(m => m.id === messageId);
    if (index !== -1) {
      const message = this.messages[index];
      message.status = 'ACKED';
      message.acked_at = Date.now();

      console.log(`[MessageQueue] ✅ 消息已确认(ACK): ${messageId}`);

      // 从活跃队列中移除（可选：保留历史记录用于调试）
      this.messages.splice(index, 1);
      this._markDirty();

      return true;
    }
    console.warn(`[MessageQueue] ⚠️  未找到消息ID: ${messageId}`);
    return false;
  }

  /**
   * markAsFailed - 标记消息发送失败（保持PENDING状态以便重试）
   */
  markAsFailed(messageId, error) {
    const message = this.messages.find(m => m.id === messageId);
    if (message) {
      message.status = 'PENDING'; // 重置为PENDING，下次重试
      message.last_error = error;
      message.sent_at = null; // 清除发送时间
      this._markDirty();

      console.log(`[MessageQueue] ❌ 消息发送失败: ${messageId}, 错误: ${error}`);
      return true;
    }
    return false;
  }

  /**
   * getAllMessages - 获取所有消息（用于调试）
   */
  getAllMessages() {
    return [...this.messages];
  }

  /**
   * getStats - 获取队列统计信息
   */
  getStats() {
    const pending = this.messages.filter(m => m.status === 'PENDING').length;
    const sent = this.messages.filter(m => m.status === 'SENT').length;
    const acked = this.messages.filter(m => m.status === 'ACKED').length;

    return {
      total: this.messages.length,
      pending: pending,
      sent: sent,
      acked: acked,
      max_capacity: this.maxQueueSize,
      utilization: ((this.messages.length / this.maxQueueSize) * 100).toFixed(1) + '%',
      oldest_pending: this.getOldestPendingTime(),
      storage_path: this.storagePath
    };
  }

  /**
   * clearAll - 清空所有消息（慎用！）
   */
  async clearAll() {
    this.messages = [];
    await this._saveToFile();
    console.log('[MessageQueue] 🗑️  队列已清空');
  }

  /**
   * forceSave - 强制立即保存到文件
   */
  async forceSave() {
    await this._saveToFile();
  }

  /**
   * destroy - 销毁实例（停止自动保存）
   */
  destroy() {
    if (this._saveTimer) {
      clearInterval(this._saveTimer);
      this._saveTimer = null;
    }

    // 最后一次保存
    this.forceSync();

    console.log('[MessageQueue] 💀 实例已销毁');
  }

  // ══════════════════════════════════════════════
  // 私有方法
  // ══════════════════════════════════════════════

  _sortMessages() {
    this.messages.sort((a, b) => a.created_at - b.created_at);
  }

  _getOldestPendingTime() {
    const pending = this.getPendingMessages();
    if (pending.length > 0) {
      const age = Date.now() - pending[0].created_at;
      return `${Math.floor(age / 1000)}s`;
    }
    return 'N/A';
  }

  _removeOldestPending() {
    const pendingIndex = this.messages.findIndex(m => m.status === 'PENDING');
    if (pendingIndex !== -1) {
      const removed = this.messages.splice(pendingIndex, 1)[0];
      this._markDirty();
      return removed.id;
    }
    return null;
  }

  _markDirty() {
    this._dirty = true;
  }

  _ensureStorageDirectory() {
    const dir = path.dirname(this.storagePath);
    if (!fs.existsSync(dir)) {
      fs.mkdirSync(dir, { recursive: true });
      console.log(`[MessageQueue] 📁 创建存储目录: ${dir}`);
    }
  }

  _loadFromFile() {
    try {
      if (fs.existsSync(this.storagePath)) {
        const data = fs.readFileSync(this.storagePath, 'utf8');
        const parsed = JSON.parse(data);

        this.version = parsed.version || '1.0';
        this.createdAt = parsed.createdAt || new Date().toISOString();
        this.messages = parsed.messages || [];

        console.log(`[MessageQueue] 📂 从文件加载队列: ${this.messages.length}条消息`);
      } else {
        console.log('[MessageQueue] 📝 存储文件不存在，创建新队列');
        this._saveToFile(); // 创建空文件
      }
    } catch (error) {
      console.error('[MessageQueue] ❌ 加载文件失败:', error);
      console.warn('[MessageQueue] ⚠️  使用空队列启动');

      // 备份损坏的文件
      if (fs.existsSync(this.storagePath)) {
        const backupPath = `${this.storagePath}.corrupted.${Date.now()}`;
        fs.renameSync(this.storagePath, backupPath);
        console.error(`[MessageQueue] 💾 损坏文件已备份到: ${backupPath}`);
      }
    }
  }

  async _saveToFile() {
    if (!this._dirty) {
      return; // 无需保存
    }

    if (this._lock) {
      console.warn('[MessageQueue] ⚠️  文件正在被其他操作使用，跳过本次保存');
      return;
    }

    this._lock = true;

    try {
      const data = {
        version: this.version,
        created_at: this.createdAt,
        last_saved: new Date().toISOString(),
        messages: this.messages,
        stats: this.getStats()
      };

      const jsonStr = JSON.stringify(data, null, 2);

      // 原子写入：先写临时文件，再重命名（防止写入过程中断电导致损坏）
      const tempPath = `${this.storagePath}.tmp`;
      fs.writeFileSync(tempPath, jsonStr, 'utf8');
      fs.renameSync(tempPath, this.storagePath);

      this._dirty = false;

      console.log(`[MessageQueue] 💾 队列已保存到文件: ${this.messages.length}条消息`);

    } catch (error) {
      console.error('[MessageQueue] ❌ 保存文件失败:', error);

      // 尝试恢复临时文件
      const tempPath = `${this.storagePath}.tmp`;
      if (fs.existsSync(tempPath)) {
        try {
          fs.unlinkSync(tempPath);
          console.log('[MessageQueue] 🧹 已清理临时文件');
        } catch (e) {
          console.error('[MessageQueue] ❌ 清理临时文件失败:', e);
        }
      }

    } finally {
      this._lock = false;
    }
  }

  forceSync() {
    return this._saveToFile();
  }

  _startAutoSave() {
    this._saveTimer = setInterval(() => {
      if (this._dirty) {
        this._saveToFile();
      }
    }, this.autoSaveInterval);
  }
}

module.exports = PersistentMessageQueue;
